doc/api/testing_api.rst

```diff
@@ -0,0 +1,66 @@
+*******
+testing
+*******
+
+
+:mod:`matplotlib.testing`
+=========================
+
+.. automodule:: matplotlib.testing
+   :members:
+   :undoc-members:
+   :show-inheritance:
+
+
+:mod:`matplotlib.testing.compare`
+=================================
+
+.. automodule:: matplotlib.testing.compare
+   :members:
+   :undoc-members:
+   :show-inheritance:
+
+
+:mod:`matplotlib.testing.conftest`
+==================================
+
+.. automodule:: matplotlib.testing.conftest
+   :members:
+   :undoc-members:
+   :show-inheritance:
+
+
+:mod:`matplotlib.testing.decorators`
+====================================
+
+.. automodule:: matplotlib.testing.decorators
+   :members:
+   :undoc-members:
+   :show-inheritance:
+
+
+:mod:`matplotlib.testing.determinism`
```

Comment:
This module exposes literally no public API and is really just a helper for matplotlib.tests.test_backend_{pdf,ps,svg}.  It should be renamed to something like matplotlib.tests._test_determinism_helpers.

Reply:
Ok, I've excluded it from the docs for now. It does not provide docstrings anyway. This makes it easier to rename. But that's for another PR (including a discussion if we need a deprecation first).